### PR TITLE
Update Pattern tests

### PR DESCRIPTION
**Pull request by mfunk**
Updated the Pattern tests (pretty much just Incident Analysis Tools) to the two method test case. It is now trapping ALL errors and failures in the toolbox.

This PR does NOT fix the problems it finds. It is only fixing the problem where the errors/failures were not reported to the unittest framework. This was because our overexuberant error handling and multi-method heirarchy were actually hiding errors from the unittest framework.

### CODE REVIEW

**Comment by csmoore** [comment]
We did look at this PR together and had the same issue reported [here](https://github.com/Esri/military-tools-geoprocessing-toolbox/issues/66#issuecomment-221906390) - this crash is caused (only in Pro Python?) when there is a call to print the stack trace ex: ` traceback.format_tb` when a Toolbox GP tool (model only?) throws an exception.

We tracked this down to the traceback/stack having an error/object like:
`Incident Analysis Tools.tbx   -   Line 7` - that was crashing when the traceback format tried to open the binary `.tbx` to print out "line 7".

I'm not sure how the ArcPy stacktrace is getting links to binary files in it - that should be investigated, but the suggested workaround is to capture the GP tool exception without printing the stack trace. Ex:

```
try:
  arcpy.ClusterAnalysis_iaTools(self.inputPointsFeatures, "#", outputClusterFeatures)
except:         
  self.fail('Exception in: ClusterAnalysis_iaTools')
```

File: utils/test/patterns_tests/ClusterAnalysisTestCase.py [hunk 1]
```diff
@@ -39,54 +42,46 @@
...
-        arcpy.AddMessage("Testing Cluster Analysis (Desktop).")
-        self.test_cluster_analysis(Configuration.patterns_DesktopToolboxPath)
-        
-    def test_cluster_analysis(self, toolboxPath):
-        try:
-            if Configuration.DEBUG == True: print("     ClusterAnalysisTestCase.test_cluster_analysis")
 
-            arcpy.ImportToolbox(toolboxPath, "iaTools")
-            outputClusterFeatures = os.path.join(Configuration.incidentScratchGDB, "outputClusters")
+    def test_cluster_analysis_pro(self):
```
Comment (csmoore) [inline utils/test/patterns_tests/ClusterAnalysisTestCase.py]: Before continuing with splitting up the test cases into "_pro" and "_desktop" test case versions - we should take a closer look at the source of the original problem capturing/trapping exceptions/errors

**Comment by mfunk** [comment]
@csmoore OK, how do you want that to be done?

**Comment by csmoore** [comment]
@mfunk - thanks - merging - we can potentially look at the other test framework/runner issues later.

**Pull request merged**